js/data/buffer.js

```diff
@@ -64,45 +70,53 @@ function Buffer(options) {
     this._createPushMethod();
 }
 
-// Start Buffer1 compatability code
-Buffer.prototype = {
-
-    get pos() { return this.length * this.itemSize; },
-    get index() { return this.length; },
+/**
+ * Bind this buffer to a WebGL context.
+ * @private
+ * @param gl The WebGL context
+ */
+Buffer.prototype.bind = function(gl) {
+    var type = gl[this.type];
 
-    get arrayType() { return this.type; },
+    if (!this.buffer) {
+        this.buffer = gl.createBuffer();
+        gl.bindBuffer(type, this.buffer);
+        gl.bufferData(type, this.arrayBuffer.slice(0, this.length * this.itemSize), gl.STATIC_DRAW);
```

Comment:
> Tangential: do you know why we have to copy the array buffer here? We dump it in the next line.

I guess that's because you want to only bind the actual data, not the full array (including empty capacity).